src/components/editMeasure/EditMeasure.tsx

```diff
@@ -25,14 +24,37 @@ import PopulationCriteriaWrapper from "./populationCriteria/PopulationCriteriaWr
 interface inputParams {
   id: string;
 }
+export interface RouteHandlerState {
+  canTravel: boolean;
+  pendingRoute: string;
+}
 export default function EditMeasure() {
-  const { url } = useRouteMatch();
-  const { id } = useParams<inputParams>();
+  const { id } = useParams();
   const measureServiceApi = useMeasureServiceApi();
   const { updateMeasure } = measureStore;
   const [loading, setLoading] = useState<boolean>(true);
+  let navigate = useNavigate();
+  const [routeHandlerState, setRouteHandlerState] = useState<RouteHandlerState>(
+    routeHandlerStore.state
+  );
+  useEffect(() => {
+    const subscription = routeHandlerStore.subscribe(setRouteHandlerState);
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+  const { updateRouteHandlerState } = routeHandlerStore;
 
-  const history = useHistory();
+  // make reusable component to throw anywhere we want to block navigation..
+  const blocker = useBlocker(
+    ({ currentLocation, nextLocation }) => !routeHandlerState.canTravel
```

Comment:
Do we need these currentLocation & nextLocation ? looks like they are never used

Reply:
no